### PR TITLE
Location rejection handling

### PR DESCRIPTION
**Pull request by suryanarayanswami**
@anuranjan  can you review the changes for this pr for service implementation for handling location rejections.

### CODE REVIEW

File: rejection-handler/src/main/java/org/motechproject/nms/rejectionhandler/service/impl/VillageHealthSubFacilityRejectionServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    private VillageHealthSubFacilityRejectionDataService villageHealthSubFacilityRejectionDataService;
+
+    @Override
+    public void createRejectedVillageHealthSubFacility(VillageHealthSubFacilityImportRejection villageHealthSubFacilityImportRejection) {
+        villageHealthSubFacilityRejectionDataService.create(villageHealthSubFacilityImportRejection);
+    }
+
+    @Override
+    public void saveRejectedVillageHealthSubFacility(VillageHealthSubFacilityImportRejection villageHealthSubFacilityImportRejection) {
+        if(   villageHealthSubFacilityRejectionDataService.findByUniqueCode(villageHealthSubFacilityImportRejection.getVillageCode(),villageHealthSubFacilityImportRejection.getHealthSubFacilityCode()) == null){
```
Comment (anuranjan) [inline rejection-handler/src/main/java/org/motechproject/nms/rejectionhandler/service/impl/VillageHealthSubFacilityRejectionServiceImpl.java:22]: make this call outside if clause and then check if the output is null or not.. Do the same in all other places

File: region/src/main/java/org/motechproject/nms/region/service/impl/HealthFacilityServiceImpl.java [hunk 2]
```diff
@@ -227,7 +239,54 @@ private String healthFacilityQuerySet(List<Map<String, Object>> healthFacilities
 
                         i++;
                     }
+                    else if(((Long) (0L)).equals(healthFacilityCode)){
+                        LOGGER.debug("rejection reason Location code zero");
+                        HealthFacilityImportRejection healthFacilityImportRejection = new HealthFacilityImportRejection((Long)healthFacility.get(LocationConstants.CSV_STATE_ID),(Long)healthFacility.get(LocationConstants.DISTRICT_ID),(String)healthFacility.get(LocationConstants.TALUKA_ID),(Long)healthFacility.get(LocationConstants.HEALTHBLOCK_ID),(Long) healthFacility.get(LocationConstants.HEALTHFACILITY_ID),(String) healthFacility.get(LocationConstants.HEALTHFACILITY_NAME),false,LocationRejectionReasons.LOCATION_CODE_ZERO_IN_FILE.toString());
+                        healthFacilityRejectionService.saveRejectedHealthFacility(healthFacilityImportRejection);
+                    }else if(!((Long) (0L)).equals(healthFacilityCode) && (healthFacilityName == null || healthFacilityName.trim().isEmpty())){
```
Comment (anuranjan) [inline region/src/main/java/org/motechproject/nms/region/service/impl/HealthFacilityServiceImpl.java:246]: do code formating wherever applicable.

File: region/src/main/java/org/motechproject/nms/region/service/impl/DistrictServiceImpl.java [hunk 3]
```diff
@@ -210,8 +214,47 @@ private String districtQuerySet(List<Map<String, Object>> districts, Map<String,
...
+                DistrictImportRejection districtImportRejection = new DistrictImportRejection(null,(Long) district.get(LocationConstants.DISTRICT_ID),(String) district.get(LocationConstants.DISTRICT_NAME),false,LocationRejectionReasons.PARENT_LOCATION_ID_NOT_PRESENT_IN_FILE.toString());
+                districtRejectionService.createRejectedDistrict(districtImportRejection);
             }
+//            else {
+//                DistrictImportRejection districtImportRejection = new DistrictImportRejection(null,(Long) district.get(LocationConstants.DISTRICT_ID),(String) district.get(LocationConstants.DISTRICT_NAME),false,LocationRejectionReasons.PARENT_LOCATION_ID_NOT_PRESENT_IN_FILE.toString());
+//                districtRejectionService.saveRejectedDistrict(districtImportRejection);
+//
+//            }
         }
+//        LOGGER.debug("Size of rejected districts is : " + rejectedDistricts.size() + "!!!");
```
Comment (anuranjan) [inline region/src/main/java/org/motechproject/nms/region/service/impl/DistrictServiceImpl.java:249]: pls remove all commented code.

File: rejection-handler/src/main/java/org/motechproject/nms/rejectionhandler/service/VillageRejectionService.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package org.motechproject.nms.rejectionhandler.service;
+
+import org.motechproject.nms.rejectionhandler.domain.VillageImportRejection;
+import org.springframework.stereotype.Service;
+
+public interface VillageRejectionService {
+
+    Long saveRejectedVillage(VillageImportRejection villageImportRejection);
```
Comment (anuranjan) [inline rejection-handler/src/main/java/org/motechproject/nms/rejectionhandler/service/VillageRejectionService.java:8]: Is save and create both inserting? Why two different names?

Comment (suryanarayanswami) [reply rejection-handler/src/main/java/org/motechproject/nms/rejectionhandler/service/VillageRejectionService.java:8]: a unique key is implemented for location code + state code, so if we dont recive location code in any data... we have to use create method which is not able to update records, while save works for updating if any such record exists,

**Pull request merged**